components/server/src/workspace/context-parser-service.ts

```diff
@@ -179,6 +186,16 @@ export class ContextParser {
             }
             context.checkoutLocation = config.config.checkoutLocation || context.repository.name;
             context.upstreamRemoteURI = this.buildUpstreamCloneUrl(context);
+            if (!context.warnings) {
+                context.warnings = [];
+            }
+
+            if (ExternalImageConfigFile.is(config.config.image)) {
+                if (config.config.image.externalSource.revision === "") {
```

Comment:
I assume here is the place I meant in https://github.com/gitpod-io/gitpod/pull/20345/files#r1832411675.

Reply:
Made it happen in [`fbd4b35` (#20345)](https://github.com/gitpod-io/gitpod/pull/20345/commits/fbd4b3549cd704f97342cbce04e10448040c10ef). I agree that it's better to have a special value.